jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -242,11 +246,43 @@ def process(self, input):
                 self.save_opt, readnoise_2d, gain_2d, self.algorithm,
                 self.weighting, max_cores, dqflags.pixel, suppress_one_group=self.suppress_one_group)
 
-        # Save the OLS optional fit product, if it exists
-        if opt_info is not None:
-            opt_model = create_optional_results_model(input_model, opt_info)
-            self.save_model(opt_model, 'fitopt', output_file=self.opt_name)
+            # To calculate the readnoise variance for weighting, ramp_fitting
+            # will be run a second time - this time with the DO_NOT_USE flags
+            # removed from the UNDERSAMP-flagged pixels. To reduce execution time,
+            # fitting will effectively be performed only on the UNDERSAMP-flagged
+            # pixels, by flagging all other pixels as DO_NOT_USE.
+
+            # First create an intermediate model ('_W' for weighting) to
+            # input to a 2nd ramp_fitting call.
+            input_model_W = copy.copy(input_model)
+
+            # Flag all pixels as DO_NOT_USE, then remove the DO_NOT_USE flag
+            # from the UNDERSAMP-flagged pixels.
+            gdq = input_model_W.groupdq.copy()
+            wh_undersamp = np.where(np.bitwise_and(gdq.astype(np.int32), dqflags.group['UNDERSAMP']))
```

Comment:
A check should be done to make sure a group has this flag set.  All this extra processing should be done only if the length of `wh_undersamp[0]` is greater than 0.  Otherwise, this is extra processing that doesn't add anything.

Reply:
Yes. Adding that check first.